app/src/main/java/com/google/samples/apps/sunflower/viewmodels/GalleryViewModel.kt

```diff
@@ -16,22 +16,56 @@
 
 package com.google.samples.apps.sunflower.viewmodels
 
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import androidx.paging.PagingData
 import androidx.paging.cachedIn
+import com.google.samples.apps.sunflower.data.UnsplashPhoto
 import com.google.samples.apps.sunflower.data.UnsplashRepository
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class GalleryViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
-    repository: UnsplashRepository
+    private val repository: UnsplashRepository
 ) : ViewModel() {
 
     private var queryString: String? = savedStateHandle["plantName"]
 
-    val plantPictures =
-        repository.getSearchResultStream(queryString ?: "").cachedIn(viewModelScope)
+
+    private val _plantPictures = MutableStateFlow<PagingData<UnsplashPhoto>?>(null)
+    val plantPictures: Flow<PagingData<UnsplashPhoto>> get() = _plantPictures.filterNotNull()
+
+    private val _isRefreshing = mutableStateOf(false)
+    val isRefreshing: State<Boolean> get() = _isRefreshing
```

Comment:
This state shouldn't be in the view model. Since it's UI state, it can be scoped in the composable

Reply:
so for alternate i think i have to expose a state flow for isRefreshing state and consume it over compose screen to get current loading state of data

Reply:
okay